engine/src/main/scala/cromwell/jobstore/SqlJobStore.scala

```diff
@@ -23,7 +23,6 @@ class SqlJobStore(sqlDatabase: EngineSqlDatabase) extends JobStore {
     val completedWorkflowIds = workflowCompletions.toList.map(_.workflowId.toString)
     for {
       _ <- sqlDatabase.addJobStores(jobCompletions map toDatabase, batchSize)
-      _ <- completedWorkflowIds traverse sqlDatabase.removeWorkflowStoreEntry
```

Comment:
Before this changeset, the `SqlJobStore` was in charge of removing entries from the workflow store. This is a little odd (job store accessing workflow store) but basically fine.

In the new coordinated version of the world, removing the store entries now [takes place in the higher-level `JobStoreWriterActor`](https://github.com/broadinstitute/cromwell/blob/040b0665f36b9b709eda5b602280f0d8de08c9e1/engine/src/main/scala/cromwell/jobstore/JobStoreWriterActor.scala#L41-L43). (If you follow through the `deleteFromStore` method it eventually calls the same `removeWorkflowStoreEntry` by way of several other classes.)

What this refactor accomplishes lets us give the `JobStoreWriterActor` a [reference](https://github.com/broadinstitute/cromwell/pull/5906/files#diff-f8c853fa534127f3ec06067c00b013583937a658271236bb2613eb37c2dab8f2R25) to the `WorkflowStoreAccess` it needs to make the coordinated call.

The other obvious factoring would be to have `SqlJobStore` receive a reference to the `WorkflowStoreAccess` but I rejected it because I felt the low level SQL code should not be muddied with the higher-level `*Access` class. Similar to how `MetadataDatabaseAccess` has a `MetadataSqlDatabase` and not vice versa.